platforms.mzn

```diff
@@ -0,0 +1,65 @@
+%Inbuilt header file
+include "table.mzn";
+
+%Thresholds of properties
+int: min_force_sensor;  
+int: min_tactile_sensor_count; 
+int: min_memory_fused;               
+
+set of PLATFORMS: platforms;
+
+%Indices of feature of every platform
+set of int: FEATURES = 1..6;
+int: name = 1;                                                              
+int: force_sensor = 2;int: tactile_sensor = 3;int: memory_availability = 4; 
+int: force_sensor_type = 5; int:speed = 6;
+ 
+%Number of platforms
+int: n_platforms;
+set of int:PLATFORMS = 1..n_platforms;
+
+%Array of platforms and properties
+array[PLATFORMS] of string: n;       
+array[PLATFORMS,FEATURES] of int: dd;
+
+%Name of the platforms
+n = ["nvidia_force","amd_tactile","intel_fused","nxp_force"];
+
+%Platform properties
+dd =  [| 1,   1,   50, 400, 1, 10,   % PF1
+       | 2,   0,  100, 100, 0, 100,  % PF2
+       | 3,   0,   10, 400, 0, 1000,  % PF3
+       | 4,   1,    0,  50, 2, 10,|];  % PF4
+       
+%Platforms to be assigned for every sensor
+array[FEATURES] of var int: force_platform; 
+array[FEATURES] of var int: tactile_platform;
+array[FEATURES] of var int: fused_platform;  
+
+%Constraint of platform names to be in the provided table
+constraint force_platform[name] in platforms;  
+constraint tactile_platform[name] in platforms;
+constraint fused_platform[name] in platforms;  
+
+%Constraint that the values to be assigned to platform names are available in the table
+constraint table(force_platform, dd);  
+constraint table(tactile_platform, dd);
+constraint table(fused_platform, dd);  
+
+%Technical constraints
+constraint ((force_platform[force_sensor] >= min_force_sensor) /\ (force_platform[speed] >5)/\(force_platform[speed] <2000));
+constraint (force_platform[force_sensor_type] != 0);
+constraint (tactile_platform[tactile_sensor] >= min_tactile_sensor_count /\ tactile_platform[speed] >5 /\ tactile_platform[speed] <2000);
```

Comment:
Introduce threshold variables for min_latency, max_latency and force_sensor_type, instead of using magical numbers such as 5 and 2000